index.bs

```diff
@@ -698,9 +698,22 @@ Native hit test result {#native-hit-test-result-section}
 
 <dfn lt="native hit test result|native hit test results">Native hit test results</dfn> returned from XR device should contain the position of the intersection point with user's environment. Depending on the native entity type and the information available to the XR device, the result should also contain orientation defined in such a way to allow the user agent to compute a surface normal to the user's environment at the intersection point.
 
-The information about position and orientation of the intersection point should be contained in [=native hit test result=]'s native origin. Native origin defines a new coordinate system in such a way that its Y axis represents the surface's normal vector at the intersection point. If the orientation is not returned from the XR device, the user agent SHOULD set the native origin in such a way that Y axis of the coordinate system it defines is pointing up (towards negative gravity vector).
+The information about position and orientation of the intersection point should be contained in [=native hit test result=]'s native origin. Native origin defines a new coordinate system in such a way that its Y axis represents the surface's normal vector at the intersection point. If the orientation is not returned from the XR device, the user agent SHOULD set the native origin in such a way that Y axis of the coordinate system it defines is pointing up (towards negative gravity vector). Additionally, the Z axis of the returned native origin should be set so that it points towards the origin of the ray used to perform the native hit test (i.e. the ray's origin lies on the YZ plane). The exact steps to obtain the native origin are described in [=calculate the native origin=] algorithm.
 
-Issue: Decide if we need to specify other axes of the coordinate system defined by hit test result's native origin to maintain compatibility between different implementations & differrent AR frameworks.
+<div class="algorithm" data-algorithm="calculate-native-origin">
+
+To <dfn>calculate the native origin</dfn> given [=native hit test result=] |result| and {{XRRay}} |offsetRay|, the user agent MUST run the following steps:
+    1. Let |coord| be a coordinate system defined by |offsetRay|'s {{XRRay/matrix}}. Let |X|, |Y|, and |Z| be the axes of coordinate system |coord|.
```

Comment:
In general we talk in terms of position and rotation transforms, do you have a way of framing this in those terms, using cross products and operations like "rotate _vector_ by _rotation_"?

I'd eventually like to write a separate geometry primitives spec so that we can talk about these operations properly, but until then I am wary of introducing _new_ operations like "let X be the coordinate system defined by matrix".

For example, here `|Z|` is simply "rotate `(0, 0, -1)` by _rotation_"

Reply:
Actually, `|Z|` is simply the `direction` of the ray :smile:

Reply:
Let me try to make it a bit more precise, I'll flip the PR into "draft" for the time being (I've noticed that I should probably be talking about angles between vector and its projection onto some plane instead of angles between planes to make this work correctly). I _think_ `|-Z|` is the direction of the ray, based on how we construct the ray's matrix.

Reply:
Reworked the algorithm a bit, should be equivalent to what was previously here.